src/main/java/de/maxhenkel/pipez/blocks/tileentity/types/ItemPipeType.java

```diff
@@ -162,17 +161,17 @@ protected void insertOrdered(PipeLogicTileEntity tileEntity, Direction side, Lis
                 if (count == 0) {
                     continue;
                 }
-                if (nonFittingItems.contains(item)) {
+                if (nonFittingItems.contains(new Tuple<>(item, tag))) {
```

Comment:
Whats the performance of this? Thats basically the same as just having an ItemStack, since an ItemStack just holds Item+NBT. I currently don't see any difference over just using ItemStack?
This also isn't very future proof, as ItemStacks won't hold NBT in 1.20.5.

Reply:
> as ItemStacks won't hold NBT in 1.20.5
ugh. this is really bad.

I mean the biggest speedup was simply through the HashMap, but after that half of the runtime was copying the `ItemStacks`. See this profile i did with only the `HashMap` changes and still with the first extractItem call: https://spark.lucko.me/lR2Bi1ovpR?hl=383